tools/common.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+DOWNLOAD=download
+UNPACK=vendor
+INSTALL=install
+
+function update_dep_from_pypi {
+    local dep=$1
+    local version=$2
+
+    mkdir -p $DOWNLOAD
+
+    local expected=$DOWNLOAD/${dep}-${version}*
+
+    if [ ! -f $expected_file ]
+    then
+        # Remove any previous versions that might be hanging around
+        rm -f $DOWNLOAD/${dep}-[0-9]*
+
+        pip download \
+            -d $DOWNLOAD \
+            --no-deps \
+            --no-binary :all: \
+            ${dep}===${version}
+    fi
+
+    unpack_sdist ${dep} ${version}
+}
+
+function unpack_sdist {
+    local dep=$1
+    local version=$2
+
+    mkdir -p $UNPACK
+
+    local expected=$UNPACK/${dep}-${version}
+
+    if [ -d $expected ]
+    then
+        echo "Not overwriting $expected"
+        return 0
+    fi
+
+    for sdist in $DOWNLOAD/${dep}-*.tar.*
+    do
+        if [[ $sdist =~ \.gz ]]; then
+            zopt=z
+        elif [[ $sdist =~ \.bz2 ]]; then
+            zopt=j
+        else
+            echo "Unknown file format $sdist"
+            return 1
+        fi
+
+        tar --directory $UNPACK \
+            -x -v -${zopt} \
+            -f $sdist \
+            --unlink-first --recursive-unlink
+    done
+}
+
+function update_dep_from_git {
+    local dep=$1
+    local repo=$2
+    local ref=$3
+
+    mkdir -p $UNPACK
+
+    # Remove any versioned copies of the dep, in case we're switching
+    # from PyPI to git.
+    rm -rf $UNPACK/${dep}-[0-9]*
+
+    # Remove any unversioned copies of the dep
+    rm -rf $UNPACK/${dep}
+
+    git clone ${repo} $UNPACK/$dep
+    (cd $UNPACK/${dep} && git checkout ${ref})
```

Comment:
Instead of including the whole git tree, we probably want to check it out to /tmp and build an sdist that we can then unpack with the previous function. That would make the treatment of both sets of sources consistent, and give us a real version number.